packages/analytics/client/src/analytics_client/analytics_client.ts

```diff
@@ -267,7 +267,11 @@ export class AnalyticsClient implements IAnalyticsClient {
     // Observer that will emit when both events occur: the OptInConfig is set + a shipper has been registered
     const configReceivedAndShipperReceivedObserver$ = combineLatest([
       this.optInConfigWithReplay$,
-      this.shipperRegistered$,
+      merge([
+        this.shipperRegistered$,
+        // Merging shipperRegistered$ with the optInConfigWithReplay$ when optedIn is false, so that we don't need to wait for the shipper if opted-in === false
+        this.optInConfigWithReplay$.pipe(filter((cfg) => cfg?.isOptedIn() === false)),
+      ]),
```

Comment:
We need this `merge` because otherwise `this.shipperRegistered$` will leave this emission hanging if never got a shipper registered. If optedIn false, we are OK to flush the queue.